lib/appium.js

```diff
@@ -490,6 +484,30 @@ class AppiumDriver extends BaseDriver {
     }
   }
 
+  async deleteAllSessions (opts = {}) {
+    if (_.isEmpty(this.sessions)) {
+      log.debug(`There are no active sessions`);
+      return;
+    }
+
+    const {
+      force = false,
+      reason = null,
+    } = opts;
+    log.debug(`Cleaning up ${_.size(this.sessions)} active session` +
+      (_.size(this.sessions) === 1 ? '' : 's'));
+    const cleanupPromises = force
+      ? _.values(this.sessions).map((drv) => drv.startUnexpectedShutdown(reason && new Error(reason)))
+      : _.keys(this.sessions).map((id) => this.deleteSession(id));
+    for (const cleanupPromise of cleanupPromises) {
+      try {
+        await cleanupPromise;
+      } catch (e) {
+        log.debug(e);
```

Comment:
do we want to exit with code 1 in this case as well? right now any errors during cleanup are ignored

Reply:
I think if this is going to do that, it ought to note the error and then return whether it errored or not, so all the promises can be awaited.

Reply:
we actuallydon't care about the result of each session deletion promise. This whole method is just to make sure there are no pending leftovers